src/fast-path.js

```diff
@@ -305,14 +309,10 @@ FPp.needsParens = function(assumeExpressionContext) {
         return true;
       }
 
-      if (
-        node.operator === "instanceof" &&
-        parent.type === "ArrowFunctionExpression"
-      ) {
-        return true;
-      }
+      if (isBinarishOpInArrowFunction(node, parent)) return true;
 
     case "LogicalExpression":
+      if (isBinarishOpInArrowFunction(node, parent)) return true;
```

Comment:
don't add this on every single one of those. Instead, add a if condition at the top with

```js
if (parent.type === 'ArrowFunctionExpression' && parent.body === node &&
  // all the conditions where it's true
) { 
  return true;
}